### PR TITLE
adding 2nd bunch of nf-core deps from update/nextflow-tools

### PR DESCRIPTION
**Pull request by marcodelapierre**
Fourth chunk of updates from #34466.

@adamjstewart

Note you had started to review these recipes in the PR above.

Review BEFORE #34563

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the 25 sha256.

**Comment by tldahlgren** [comment]
Adding @adamjstewart as reviewer since these are all python packages.

File: var/spack/repos/builtin/packages/py-a2wsgi/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class PyA2wsgi(PythonPackage):
+    """Convert WSGI app to ASGI app or ASGI app to WSGI app."""
+
+    homepage = "https://github.com/abersheeran/a2wsgi"
+    pypi = "a2wsgi/a2wsgi-1.6.0.tar.gz"
+
+    version("1.6.0", sha256="67a9902db6da72c268a24d4e5d01348f736980a577279b7df801c8902aba8554")
+
+    depends_on("python@3.6.2:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-a2wsgi/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-apispec/package.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+class PyApispec(PythonPackage):
+    """A pluggable API specification generator."""
+
+    homepage = "https://github.com/marshmallow-code/apispec"
+    pypi = "apispec/apispec-6.0.2.tar.gz"
+
+    version("6.0.2", sha256="e76d80b739edef4be213092a6384ad7fd933ba7d64f6d5a0aff8d4da1bef7887")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-apispec/package.py]:
```suggestion
    depends_on("python@3.7:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-bagit-profile/package.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PyBagitProfile(PythonPackage):
+    """A simple Python module for validating BagIt profiles."""
+
+    homepage = ""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bagit-profile/package.py]:
```suggestion
    homepage = "https://github.com/bagit-profiles/bagit-profiles-validator"
```

File: var/spack/repos/builtin/packages/py-bagit-profile/package.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    pypi = "bagit-profile/bagit_profile-1.3.1.tar.gz"
+
+    version("1.3.1", sha256="57798cdcf98b32a413edb29382d85f4f8c44d3204940d7e12d84998521a98c3f")
+
+    depends_on("python@3.4:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-bagit@0.9.8:", type=("build", "run"))
+    depends_on("py-requests@0.14.2:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bagit-profile/package.py]:
```suggestion
    depends_on("py-bagit", type=("build", "run"))
    depends_on("py-requests", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-bagit-profile/package.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+class PyBagitProfile(PythonPackage):
+    """A simple Python module for validating BagIt profiles."""
+
+    homepage = ""
+    pypi = "bagit-profile/bagit_profile-1.3.1.tar.gz"
+
+    version("1.3.1", sha256="57798cdcf98b32a413edb29382d85f4f8c44d3204940d7e12d84998521a98c3f")
+
+    depends_on("python@3.4:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bagit-profile/package.py]:
```suggestion
    depends_on("python@2.7,3.4:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-bdbag/package.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    """The bdbag utilities are a collection of software programs for working
+    with BagIt packages that conform to the BDBag and Bagit/RO profiles.
+    """
+
+    homepage = "https://github.com/fair-research/bdbag/"
+    pypi = "bdbag/bdbag-1.6.3.tar.gz"
+
+    version("1.6.3", sha256="1ad2e4956045cb3d43a6276391ad919e42a90a2443727dbc5b1ac6eeb6d6e3c9")
+
+    depends_on("python@3.5:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bdbag/package.py]:
```suggestion
    depends_on("python@2.7:2,3.5:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-bdbag/package.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools-scm@:5.999999", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bdbag/package.py]:
```suggestion
    depends_on("py-setuptools-scm@:5", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-bdbag/package.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+    depends_on("py-pytz", type=("build", "run"))
+    depends_on("py-tzlocal@2.1", type=("build", "run"))
+    depends_on("py-certifi", type=("build", "run"))
+    depends_on("py-requests@2.7:2.25.1", when="^python@3.5", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bdbag/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-bioblend/package.py [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    homepage = "https://bioblend.readthedocs.io"
+    pypi = "bioblend/bioblend-1.0.0.tar.gz"
+
+    version("1.0.0", sha256="3794288bbf891ae6edc1bcdd9618a3ae16b6ed4a04c946505f7e29f2f28898a5")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-requests@2.20.0:", type=("build", "run"))
+    depends_on("py-requests-toolbelt@0.5.1:0.8.999999,0.9.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bioblend/package.py]:
```suggestion
    depends_on("py-requests-toolbelt@0.5.1:0.8,0.9.1:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-circus/package.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    """
+
+    homepage = "https://github.com/circus-tent/circus"
+    pypi = "circus/circus-0.18.0.tar.gz"
+
+    version("0.18.0", sha256="193ce8224e068ced66724cf483106fb6674b51a57583ac1a0e7ed7a7ee8c71ab")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-flit-core@3.4:3.999999", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-circus/package.py]:
```suggestion
    depends_on("py-flit-core@3.4:3", type="build")
```

File: var/spack/repos/builtin/packages/py-circus/package.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+    multiple processes and sockets.
+    """
+
+    homepage = "https://github.com/circus-tent/circus"
+    pypi = "circus/circus-0.18.0.tar.gz"
+
+    version("0.18.0", sha256="193ce8224e068ced66724cf483106fb6674b51a57583ac1a0e7ed7a7ee8c71ab")
+
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-circus/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-cloudbridge/package.py [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+class PyCloudbridge(PythonPackage):
+    """A simple layer of abstraction over multiple cloud providers."""
+
+    homepage = "http://cloudbridge.cloudve.org"
+    pypi = "cloudbridge/cloudbridge-3.1.0.tar.gz"
+
+    version("3.1.0", sha256="f9d3c1ae36b14a1c953d36c21a35fa2c72d42831cbbfe6117d13b25e9cccb28c")
+
+    depends_on("python@3.4:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cloudbridge/package.py:19]:
```suggestion
    depends_on("py-setuptools", type="build")
    depends_on("py-nose@1:", type="build")
```

File: var/spack/repos/builtin/packages/py-cloudbridge/package.py [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+
+    depends_on("py-six@1.11:", type=("build", "run"))
+    depends_on("py-tenacity@6.0:", type=("build", "run"))
+    depends_on("py-deprecation@2.0.7:", type=("build", "run"))
+    depends_on("py-pyeventsystem@:1.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cloudbridge/package.py]:
```suggestion
    depends_on("py-pyeventsystem@:1", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwl-upgrader/package.py [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+
+class PyCwlUpgrader(PythonPackage):
+    """Common Workflow Language standalone document upgrader"""
+
+    homepage = "https://github.com/common-workflow-language/cwl-upgrader"
+    pypi = "cwl-upgrader/cwl-upgrader-1.2.4.tar.gz"
+
+    version("1.2.4", sha256="b25fc236407343d44cc830ac3f63eed395b8d872fc7e17db92cde583d4a3b2ec")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwl-upgrader/package.py]:
```suggestion
    depends_on("python@3.6:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwl-upgrader/package.py [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-ruamel-yaml@0.16.0:0.17.22", when="^python@3.10:", type=("build", "run"))
+    depends_on("py-ruamel-yaml@0.15.98:0.17.22", when="^python@3.9:", type=("build", "run"))
+    depends_on("py-ruamel-yaml@0.15.78:0.17.22", when="^python@3.8:", type=("build", "run"))
+    depends_on("py-ruamel-yaml@0.15.71:0.17.22", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwl-upgrader/package.py]:
```suggestion
    depends_on("py-ruamel-yaml@0.16.0:0.17.21", when="^python@3.10:", type=("build", "run"))
    depends_on("py-ruamel-yaml@0.15.98:0.17.21", when="^python@3.9:", type=("build", "run"))
    depends_on("py-ruamel-yaml@0.15.78:0.17.21", when="^python@3.8:", type=("build", "run"))
    depends_on("py-ruamel-yaml@0.15.71:0.17.21", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwl-utils/package.py [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+    """
+
+    homepage = "https://github.com/common-workflow-language/cwl-utils"
+    pypi = "cwl-utils/cwl-utils-0.21.tar.gz"
+
+    version("0.21", sha256="583f05010f7572f3a69310325472ccb6efc2db7f43dc6428d03552e0ffcbaaf9")
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-pytest-runner", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwl-utils/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-cwl-utils/package.py [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+
+    depends_on("py-cwl-upgrader@1.2.3:", type=("build", "run"))
+    depends_on("py-packaging", type=("build", "run"))
+    depends_on("py-rdflib", type=("build", "run"))
+    depends_on("py-requests", type=("build", "run"))
+    depends_on("py-cachecontrol", type=("build", "run"))
+    depends_on("py-schema-salad@8.3.20220825114525:8.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwl-utils/package.py]:
```suggestion
    depends_on("py-schema-salad@8.3.20220825114525:8", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+    )
+    version(
+        "3.1.20211107152837",
+        sha256="ae1cd4626b5330457b1a62bcb2580f36f530264a80222f2cc17cf65899ebf04e",
+    )
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-requests@2.4.3:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
    depends_on("py-requests@2.6.1:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 19]
```diff
@@ -0,0 +1,49 @@
...
+    version(
+        "3.1.20221109155812",
+        sha256="82676ea315ce84fc4057d92c040af15dde3e897527ea4ae70c1033b0eca20c2a",
+    )
+    version(
+        "3.1.20211107152837",
+        sha256="ae1cd4626b5330457b1a62bcb2580f36f530264a80222f2cc17cf65899ebf04e",
+    )
+
+    depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
    depends_on("python@3.6:3", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("py-setuptools", type="build")
+
+    depends_on("py-requests@2.4.3:", type=("build", "run"))
+    depends_on("py-ruamel-yaml@0.16.0:0.17.21", when="^python@3.10:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("py-ruamel-yaml@0.15:0.17.21", type=("build", "run"))
+    depends_on("py-rdflib@4.2.2:6.2.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
    depends_on("py-rdflib@4.2.2:6.2", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("py-rdflib@4.2.2:5.999999", when="^python@:3.6", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("py-shellescape@3.4.1:3.8.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
    depends_on("py-shellescape@3.4.1:3.8", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 24]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("py-schema-salad@8.2.20211104054942:8.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
    depends_on("py-schema-salad@8.2.20211104054942:8", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 25]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("py-prov@1.5.1", type=("build", "run"))
+    depends_on("py-bagit@1.8.1", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
    depends_on("py-bagit@1.6.4:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("py-mypy-extensions", type=("build", "run"))
+    depends_on("py-psutil@5.6.6:", type=("build", "run"))
+    depends_on("py-typing-extensions", type=("build", "run"))
+    depends_on("py-coloredlogs", type=("build", "run"))
+    depends_on("py-pydot@1.4.1:", type=("build", "run"))
+    depends_on("py-argcomplete@1.12.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
    depends_on("py-argcomplete", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cwltool/package.py [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+    depends_on("py-pyparsing@:3.0.1,3.0.3:", type=("build", "run"))
+    depends_on("py-pyparsing@:3", when="^python@:3.6", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cwltool/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-schema-salad/package.py [hunk 28]
```diff
@@ -0,0 +1,37 @@
...
+    homepage = "https://github.com/common-workflow-language/schema_salad"
+    pypi = "schema-salad/schema-salad-8.3.20221209165047.tar.gz"
+
+    version(
+        "8.3.20221209165047",
+        sha256="d97cc9a4d7c4255eb8000bcebaa8ac0d1d31801c921fd4113ab3051c1e326c7c",
+    )
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-schema-salad/package.py]:
```suggestion
    depends_on("py-setuptools@45:", type="build")
```

File: var/spack/repos/builtin/packages/py-schema-salad/package.py [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+
+    depends_on("py-ruamel-yaml@0.17.6:0.17.21", when="^python@3.7:", type=("build", "run"))
+    depends_on("py-ruamel-yaml@0.16.12:0.17.21", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-schema-salad/package.py]:
```suggestion
    depends_on("py-ruamel-yaml@0.17.6:0.17.21", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-schema-salad/package.py [hunk 30]
```diff
@@ -0,0 +1,37 @@
...
+    depends_on("py-rdflib@4.2.2:6.999999", type=("build", "run"))
+    depends_on("py-rdflib@4.2.2:5.999999", when="^python@:3.6", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-schema-salad/package.py]:
```suggestion
    depends_on("py-rdflib@4.2.2:6", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-schema-salad/package.py [hunk 31]
```diff
@@ -0,0 +1,37 @@
...
+    depends_on("py-rdflib-jsonld@0.4.0:0.6.0", when="^python@:3.6", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-schema-salad/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-schema-salad/package.py [hunk 32]
```diff
@@ -0,0 +1,37 @@
...
+    depends_on("py-mistune@2.0.3:2.0.999999", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-schema-salad/package.py]:
```suggestion
    depends_on("py-mistune@2.0.3:2.0", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-schema-salad/package.py [hunk 33]
```diff
@@ -0,0 +1,37 @@
...
+    depends_on("py-cachecontrol@0.12.11+filecache", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-schema-salad/package.py]:
```suggestion
    depends_on("py-cachecontrol@0.11.7:0.12+filecache", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-tuspy/package.py [hunk 34]
```diff
@@ -0,0 +1,25 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class PyTuspy(PythonPackage):
+    """A Python client for the tus resumable upload protocol -> http://tus.io"""
+
+    homepage = ""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tuspy/package.py]:
```suggestion
    homepage = "http://github.com/tus/tus-py-client/"
```

File: var/spack/repos/builtin/packages/py-schema-salad/package.py [hunk 35]
```diff
@@ -0,0 +1,37 @@
...
+    pypi = "schema-salad/schema-salad-8.3.20221209165047.tar.gz"
+
+    version(
+        "8.3.20221209165047",
+        sha256="d97cc9a4d7c4255eb8000bcebaa8ac0d1d31801c921fd4113ab3051c1e326c7c",
+    )
+
+    depends_on("python@3.6:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-schema-salad/package.py]:
```suggestion

    depends_on("py-requests@1:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-schema-salad/package.py [hunk 36]
```diff
@@ -0,0 +1,37 @@
...
+    depends_on("py-rdflib-jsonld@0.4.0:0.6.0", when="^python@:3.6", type=("build", "run"))
+    depends_on("py-mistune@2.0.3:2.0.999999", type=("build", "run"))
+    depends_on("py-cachecontrol@0.12.11+filecache", type=("build", "run"))
+
+    depends_on("py-black@19.10b0:22.12", type=("build", "run"))
+
+    depends_on("py-mypy@0.991", type=("build", "run"))
+    depends_on("py-types-pkg-resources", type=("build", "run"))
+    depends_on("py-types-requests", type=("build", "run"))
+    depends_on("py-types-dataclasses", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-schema-salad/package.py]:
```suggestion
    depends_on("py-setuptools-scm@6.2:+toml", type="build")
    depends_on("py-mypy@0.991", type="build")
    depends_on("py-black@19.10b0:", type="build")
    depends_on("py-types-pkg-resources", type="build")
    depends_on("py-types-requests", type="build")
    depends_on("py-types-dataclasses", type="build")
    depends_on("py-types-setuptools", type="build")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**